### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Limits.Preserves.Shapes.Pullbacks

### PR DESCRIPTION
**Pull request by mattrobball**
---

- [x] depends on: #2522
[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2522
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/CategoryTheory/Limits/Preserves/Shapes/Pullbacks.lean [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+  (IsLimit.postcomposeHomEquiv (diagramIsoCospan.{v₂} _) _).symm.trans <|
+    IsLimit.equivIsoLimit <|
+      Cones.ext (Iso.refl _) <| by
+        rintro (_ | _ | _) <;> dsimp <;> simp only [comp_id, id_comp, G.map_comp]
+#align category_theory.limits.is_limit_map_cone_pullback_cone_equiv CategoryTheory.Limits.isLimitMapConePullbackConeEquiv
+
+/-- The property of preserving pullbacks expressed in terms of binary fans. -/
+def isLimitPullbackConeMapOfIsLimit [PreservesLimit (cospan f g) G]
+    (l : IsLimit (PullbackCone.mk h k comm)) : 
+    -- Porting note: help figure out the hole
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Preserves/Shapes/Pullbacks.lean:67]: Please make this note a bit more explicit. I don't understand it.

File: Mathlib/CategoryTheory/Limits/Preserves/Shapes/Pullbacks.lean [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+    have : G.map h ≫ G.map f = G.map k ≫ G.map g := by rw [←G.map_comp,←G.map_comp,comm] 
+    IsLimit (PullbackCone.mk (G.map h) (G.map k) this) :=
+  isLimitMapConePullbackConeEquiv G comm (PreservesLimit.preserves l)
+#align category_theory.limits.is_limit_pullback_cone_map_of_is_limit CategoryTheory.Limits.isLimitPullbackConeMapOfIsLimit
+
+/-- The property of reflecting pullbacks expressed in terms of binary fans. -/
+def isLimitOfIsLimitPullbackConeMap [ReflectsLimit (cospan f g) G]
+    -- Porting note: more typeclass help
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Preserves/Shapes/Pullbacks.lean:75]: In what sense?

File: Mathlib/CategoryTheory/Limits/Preserves/Shapes/Pullbacks.lean [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+    from by simp only [←G.map_comp,comm]))) : IsLimit (PullbackCone.mk h k comm) :=
+  ReflectsLimit.reflects ((isLimitMapConePullbackConeEquiv G comm).symm l)
+#align category_theory.limits.is_limit_of_is_limit_pullback_cone_map CategoryTheory.Limits.isLimitOfIsLimitPullbackConeMap
+
+variable (f g) [PreservesLimit (cospan f g) G]
+
+/-- If `G` preserves pullbacks and `C` has them, then the pullback cone constructed of the mapped
+morphisms of the pullback cone is a limit. -/
+def isLimitOfHasPullbackOfPreservesLimit [i : HasPullback f g] :
+    -- Porting note: more typeclass help
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Limits/Preserves/Shapes/Pullbacks.lean:86]: Idem.

File: Mathlib/CategoryTheory/Limits/Preserves/Shapes/Pullbacks.lean (hunk 1 above)
Comment (mattrobball) [reply Mathlib/CategoryTheory/Limits/Preserves/Shapes/Pullbacks.lean:67]: Oh. This one probably doesn't rise to the level of a porting note. I had to fill in `h` and `k` similarly for the others.

Comment (jcommelin) [reply Mathlib/CategoryTheory/Limits/Preserves/Shapes/Pullbacks.lean:67]: Ok, in that case, please remove the note (-;

Comment (mattrobball) [reply Mathlib/CategoryTheory/Limits/Preserves/Shapes/Pullbacks.lean]: Removed

File: Mathlib/CategoryTheory/Limits/Preserves/Shapes/Pullbacks.lean (hunk 2 above)
Comment (mattrobball) [reply Mathlib/CategoryTheory/Limits/Preserves/Shapes/Pullbacks.lean]: Removed

File: Mathlib/CategoryTheory/Limits/Preserves/Shapes/Pullbacks.lean (hunk 3 above)
Comment (mattrobball) [reply Mathlib/CategoryTheory/Limits/Preserves/Shapes/Pullbacks.lean]: Removed

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**